### PR TITLE
[PatternLang] Add a relay LetPattern

### PR DESCRIPTION
**Pull request by mbrookhart**
I found myself needing a Let pattern for some object detection work. Thanks for taking a look!

### CODE REVIEW

File: python/tvm/relay/dataflow_pattern/__init__.py [hunk 1]
```diff
@@ -337,6 +337,29 @@ def is_if(cond, true_branch, false_branch):
     return IfPattern(cond, true_branch, false_branch)
 
 
+def is_let(var, value, body):
+    """
+    Syntatic sugar for creating an IfPattern.
```
Comment (comaniac) [inline python/tvm/relay/dataflow_pattern/__init__.py:342]:
```suggestion
    Syntatic sugar for creating a LetPattern.
```

File: python/tvm/relay/dataflow_pattern/__init__.py [hunk 2]
```diff
@@ -579,6 +602,27 @@ def __init__(self, cond: "DFPattern", true_branch: "DFPattern", false_branch: "D
         self.__init_handle_by_constructor__(ffi.IfPattern, cond, true_branch, false_branch)
 
 
+@register_df_node
+class LetPattern(DFPattern):
+    """A patern matching a Relay If.
```
Comment (comaniac) [inline python/tvm/relay/dataflow_pattern/__init__.py:607]:
```suggestion
    """A patern matching a Relay Let.
```

**Review by comaniac — APPROVED** [review]
LGTM. Just a few typos.

**Comment by masahi** [comment]
I wonder, do we need to match against the let bound variable? Isn't it always `Var`?

**Comment by mbrookhart** [comment]
@masahi I agree the Expr will always be a var, but I wanted to let users match on more complicated attributes like name/style/shape/etc, so I left it a general in pattern

**Review by masahi — APPROVED** [review]

**Comment by masahi** [comment]
Thanks @mbrookhart @comaniac

**Pull request merged**